framework/src/Volo.Abp.Core/Volo/Abp/Check.cs

```diff
@@ -162,4 +162,182 @@ public static ICollection<T> NotNullOrEmpty<T>(ICollection<T> value, [InvokerPar
 
         return value;
     }
+
+    public static Int16 NotNegativeOrZero(
+        Int16 type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if(type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if(type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+    public static Int32 NotNegativeOrZero(
+        Int32 type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if (type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if (type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+    public static Int64 NotNegativeOrZero(
+        Int64 type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if (type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if (type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+    public static float NotNegativeOrZero(
+        float type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if (type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if (type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+    public static double NotNegativeOrZero(
+        double type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if (type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if (type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+    public static decimal NotNegativeOrZero(
+        decimal type,
+        [InvokerParameterName][NotNull] string parameterName)
+    {
+        if (type == 0)
+        {
+            throw new ArgumentException($"{parameterName} is equal to zero");
+        }
+        else if (type < 0)
+        {
+            throw new ArgumentException($"{parameterName} is less than zero");
+        }
+        return type;
+    }
+
+
+
+    public static Int16 WithinRange(
```

Comment:
Please rename all to `Range`

Reply:
I have implemented the change  in this commit: [aa08d0d](https://github.com/abpframework/abp/pull/12576/commits/aa08d0d5ad86e298e433c15c26a82a36f8a173b9)